### PR TITLE
Hopefully fixed the issues with editing water intervals

### PR DESCRIPTION
**Pull request by ChrisKurz098**

### CODE REVIEW

File: public/javascript/edit-plant.js [hunk 1]
```diff
@@ -35,7 +35,8 @@ const userSelectWfMult = document.querySelector("#wfMult").value.trim();
       console.log("plant updated");
       document.location.replace("/");
     } else {
-      alert("Plant already exists in account!");
+      //if no change, go to dashboard
+      document.location.replace("/");
```
Comment (emdok) [inline public/javascript/edit-plant.js:39]: nice, yes this makes sense. TY!

File: utils/helpers.js [hunk 2]
```diff
@@ -7,11 +7,15 @@ module.exports = {
     .toLowerCase();
   },
 
-  waterFreq: (interval, dayFlag) => {
-    const day = Math.round(interval % 7);
-    const week = Math.round(interval / 7);
+  waterFreq: (interval, dayFlag, weeksAsDays) => {
+    const day = Math.floor(interval % 7) ;
+    let week = Math.floor(interval/7);
+    console.log('Day: ',day,"Weeks: ", week)
```
Comment (emdok) [inline utils/helpers.js:13]: Can we remove this console.log?

File: utils/helpers.js [hunk 3]
```diff
@@ -7,11 +7,15 @@ module.exports = {
...
 
-    return (dayFlag)? day : week;
+    if (weeksAsDays) week = week*7;
```
Comment (emdok) [inline utils/helpers.js:15]: could make this one a ternary as well
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

**Review by emdok — CHANGES_REQUESTED** [review]
Left two very small nits, then this is good to go!

**Comment by ChrisKurz098** [comment]
OK. I fixed those issues and then some. There were some security vulnerabilities where any user could send an api request to delete or update any user. Now only if the session user id and requested user id match can a user be updated. Also, now only the admin can delete a user. These two fixes stop someone from being able to update or delete the user admin or update another users info

**Review by emdok — APPROVED** [review]

**Pull request merged**